deploy/operator/setup_hive.sh

```diff
@@ -16,7 +46,57 @@ spec:
   sourceNamespace: openshift-marketplace
 EOCR
 
-wait_for_operator "hive-operator" "openshift-operators"
-wait_for_crd "clusterdeployments.hive.openshift.io"
+  wait_for_operator "hive-operator" "openshift-operators"
+  wait_for_crd "clusterdeployments.hive.openshift.io"
+
+  echo "Hive installed successfully!"
+}
+
+function from_upstream() {
+  HIVE_DIR="${HIVE_DIR:-${HOME}/go/src/github.com/openshift/hive}"
+
+  if [ ! -d "${HIVE_DIR}" ]; then
+    git clone https://github.com/openshift/hive.git "${HIVE_DIR}"
+  fi
+
+  pushd $HOME/go/src/github.com/openshift/hive
+  git fetch origin
+  git pull origin master
+
+  if [ "${DISCONNECTED}" = "true" ]; then
+    export IMG="${LOCAL_REGISTRY}/localimages/hive:latest"
+    oc image mirror \
+       -a ${AUTHFILE} \
+       ${HIVE_IMAGE} \
+       ${IMG}
+  else
+    export IMG="${HIVE_IMAGE}"
+  fi
+
+  make deploy
+  wait_for_pod "hive-operator" "hive" "control-plane=hive-operator"
+  wait_for_pod "hive-controllers" "hive" "control-plane=controller-manager"
+
+  popd
+}
+
+function install_dependencies() {
+  echo "Installing kustomize..."
+  (cd /usr/local/bin && curl -s "https://raw.githubusercontent.com/\
+kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | bash)
+
+  echo "Installing golang..."
+  curl -L https://storage.googleapis.com/golang/getgo/installer_linux -o /tmp/golang_installer
+  chmod u+x /tmp/golang_installer
+  /tmp/golang_installer
+  rm /tmp/golang_installer
+
+  echo "Activating go command on current shell..."
+  set +u
+  source /root/.bash_profile
+  set -u
+}
+
+declare -F $@ || (print_help && exit 1)
 
-echo "Hive installed successfully!"
+"$@"
```

Comment:
Now `deploy.sh` would fail since it's not deploying hive. You need to update the script to call `with_olm` by default

Reply:
Right, I'll fix it
It's problematic I don't usually use ``deploy.sh`` :sweat_smile: